### PR TITLE
Fix version status bugs on dashboard & publish/unpublish json 

### PR DESCRIPTION
**Pull request by gracez72**
BUGS: [BUG-34251](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/Adaptive%20Cards/Intern%20GitHub/Y20-VW/11?workitem=34251), [BUG-34322](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/Adaptive%20Cards/Intern%20GitHub/Y20-VW/11?workitem=34322)

This PR fixes the incorrect template instance state in the recently viewed table and the publish/unpublish deleting template json bug.

### CODE REVIEW

File: private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts [hunk 1]
```diff
@@ -388,14 +388,14 @@ export class TemplateServiceClient {
             if (template || state) {
               version = incrementVersion(existingTemplate);
               if (instance.state === TemplateState.deprecated) templateState = templateState === TemplateState.live ? TemplateState.live : TemplateState.draft;
-              templateInstance = setTemplateInstanceParam(templateInstance, templateData, templateState, isShareable, version);
+              templateInstance = setTemplateInstanceParam(templateInstance, templateData, templateState, isShareable, version, template? template : instance.json);
```
Comment (Jackbk) [inline private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts]: Formatting, space before ?

File: private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts [hunk 2]
```diff
@@ -388,14 +388,14 @@ export class TemplateServiceClient {
...
               templateInstances.push(templateInstance);
             }
             added = true;
             continue;
           } else if (instance.state === TemplateState.live && templateState === TemplateState.deprecated) {
             // Set the template to deprecated
-            templateInstance = setTemplateInstanceParam(templateInstance, templateData, templateState, isShareable, version);
+            templateInstance = setTemplateInstanceParam(templateInstance, templateData, templateState, isShareable, version, template? template : instance.json);
```
Comment (Jackbk) [inline private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts]: ditto

File: private-templates-service/client/src/components/Dashboard/RecentlyViewed/RecentlyViewedTable.tsx [hunk 3]
```diff
@@ -50,14 +49,8 @@ class RecentlyViewedTable extends React.Component<Props> {
...
-              <RecentlyViewedStatusIndicator
-                state={
-                  template.isLive
-                    ? PostedTemplate.StateEnum.Live
-                    : PostedTemplate.StateEnum.Draft
-                }
-              />
-              <Status>{template.isLive ? "Published" : "Draft"}</Status>
+              <RecentlyViewedStatusIndicator state={template.instances[0].state}/>
+              <Status>{template.instances[0].state && template.instances[0].state.toString().charAt(0).toUpperCase() + template.instances[0].state.toString().slice(1)}</Status>
```
Comment (Jackbk) [inline private-templates-service/client/src/components/Dashboard/RecentlyViewed/RecentlyViewedTable.tsx]: Can you extract this logic out to a const to make it easier to understand whats going on?

Comment (gracez72) [reply private-templates-service/client/src/components/Dashboard/RecentlyViewed/RecentlyViewedTable.tsx]: Moved to utils file, thanks!

**Pull request merged**